src/sbi-integration.adoc

```diff
@@ -0,0 +1,65 @@
+[#section_sbi_integration]
+== Integrating CHERI extensions with the SBI
+
+This section describes changes to integrate the SBI (supervisor binary
+interface) cite:[riscv-sbi], {cheri_base_ext_name}, and
+{cheri_legacy_ext_name}. It must be implemented in a standard-conforming
+supervisor execution environment which supports {cheri_base_ext_name} or
+{cheri_legacy_ext_name}.
+
+=== Binary Encoding
+
+Invocation of SBI functions is only allowed if <<pcc>> grants <<asr_perm>>. If
+an `ECALL` instruction is executed with <<pcc>> that does not grant
+<<asr_perm>>, the supervisor execution environment generates a CHERI exception.
+<<stval>> is set as for a forbidden CSR access.
```

Comment:
While I believe this makes sense for most calls, I think this may be overly restrictive. I don't see anything in the current space that would be particularly useful to non-ASR callers, but I think we should allow SBI to expose APIs that are available to all callers (e.g. do we really need to restrict querying the SBI version?).